### PR TITLE
llms/anthropic: Implement new messages api

### PR DESCRIPTION
**Pull request by GRVYDEV**
Addresses #695 

### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [x] Describes the source of new concepts.
- [x] References existing implementations as appropriate.
- [ ] Contains test coverage for new functions.
- [ ] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

This PR implements the new anthropic messages API in a backwards compatible way. The messages API is opt in by default. The implementation was based off of the `openai` implementation. This allows for system prompts and unlocks the path to more robust Anthropic integration in the future. It is worth noting that I did _not_ implement streaming because I did not have time and do not need it for my use case. It also does not implement multi-modal chat messages.

### CODE REVIEW

**Comment by tmc** [comment]
Converting to draft until this is ready for review. Please change it back when ready.

**Comment by sansmoraxz** [comment]
FYI I have working messages api for AWS bedrock in #666 .

**Comment by tmc** [comment]
@GRVYDEV can you bring this up to ready for review, very much want to get a nice opus-powered example using the new messages api in here!

**Comment by GRVYDEV** [comment]
Hey Travis super excitied that you want to get a demo out of this. I however, can't commit to getting this PR ready to review in a timely manner. (Recently started a company so swamped with other stuff atm). Feel free to take over this PR and make any changes needed to get it mergable. FWIW I am currently using this in production and it seems to work so just needs some cleanup.

File: llms/anthropic/internal/anthropicclient/messages_completions.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+type ChatMessage struct {
+	Role    string `json:"role"`
+	Content string `json:"content"`
```
Comment (sansmoraxz) [inline llms/anthropic/internal/anthropicclient/messages_completions.go:14]: Maybe use a struct for this. Easier to use multimodal here.

**Comment by tmc** [comment]
The trouble with open source -- your contributors are too busy starting companies! :P I may take this on.

**Comment by tmc** [comment]
Pushed some extension/cleanup including streaming support.

**Comment by tmc** [comment]
Implemented in https://github.com/tmc/langchaingo/pull/707

**Pull request closed without merging**